o2-sim: Improved signal handling

sawenzel opened the pull request:
This should improve the shutdown phase
of the distributed simulation system upon errors:

*) now terminate using process groups in
   order to prevent leaking child processes
*) improved signal broadcasting in the Geant transport
   processes (within the process group)

sawenzel committed:
o2-sim: Improved signal handling

This should improve the shutdown phase
of the distributed simulation system upon errors:

*) now terminate using process groups in
   order to prevent leaking child processes
*) improved signal broadcasting in the Geant transport
   processes (within the process group)

davidrohr commented on run/O2SimDeviceRunner.cxx:
```diff
@@ -35,20 +35,19 @@ namespace bpo = boost::program_options;
+    // forward to master
+    kill(groupid, signal);
   }
+  _exit(1); // we treat signal interruption as an error
```
I am not 100% sure, but are we sure the only signal we will receive is KILL and TERM? Perhaps we should have a filter for the singal in here?

sawenzel replied:
Why? The filter is specified further below in the main function.

davidrohr replied:
ok, sorry, I missed that. Then it is OK. Perhaps we should also forward `SIGQUIT` and `SIGKILL`? Or probably SIGKILL cannot be handled?

sawenzel replied:
SIGKILL can't be handled.

davidrohr replied:
ok, I would put that in a comment to avoid confusion, and perhaps I'd add SIGQUIT just in case it is used at some time to avoid misleading behavior.

sawenzel commented:
@MichaelLettrich @TimoWilken : MacOS has a serious problem.

TimoWilken commented:
@sawenzel thanks, looking into it.

sawenzel commented:
@ktf: There are still compiler crashes despite the Vc improvement. I think the reason is something else and we rather need to rethink our way to build / the hardware on which we build:

```/mnt/mesos/sandbox/sandbox/o2/sw/SOURCES/O2/6299/0/Analysis/Tasks/PWGHF/taskLc.cxx
c++: fatal error: Killed signal terminated program cc1plus```

davidrohr commented:
@ironMann : The FullCI fails in datadistribution probably due to the o2::byte / std::byte change:
```
/mnt/mesos/sandbox/sandbox/o2-fullci/sw/SOURCES/DataDistribution/v0.10.16/v0.10.16/src/common/SubTimeFrameFileReader.cxx: In member function 'o2::header::Stack o2::DataDistribution::SubTimeFrameFileReader::getHeaderStack(std::size_t&)':
/mnt/mesos/sandbox/sandbox/o2-fullci/sw/SOURCES/DataDistribution/v0.10.16/v0.10.16/src/common/SubTimeFrameFileReader.cxx:121:30: error: cannot convert 'unsigned char*' to 'std::byte*' in initialization
```
Could you have a look? (This ran yesterday evening)

ktf commented:
I am on it, but I still have one error when compiling it on my box.

ktf commented:
@MichaelLettrich I think you need to deploy one more mac with the updated system deps.

MichaelLettrich commented:
@ktf  just finished deploying on all macs- will take a while till the effects will show due to rebuilds.

sawenzel committed:
CcdbApi: disable token checking on MacOS

Token checking has problems in MacOS (hangs in unit tests in some environments, etc).

Given that the mechanism is not relevant on this non-production platform,
let's simply disable it.

sawenzel committed:
o2sim: Harmonize socket names; fix rm command
